frontend/src/components/DrawComponent.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Draw from 'ol/interaction/Draw.js';
+import { unByKey } from 'ol/Observable.js';
+import VectorLayer from "ol/layer/Vector";
+import VectorSource from "ol/source/Vector";
+
+const DrawComponent = ({ map, geometryType, onDeleteGeometries }) => {
```

Comment:
It is indeed a good practice to utilize component props decomposition when working with JavaScript